pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -352,15 +363,50 @@ func New(ctx context.Context, plArgs runtime.Object, fh framework.Handle, fts fe
 		classLister:                fh.SharedInformerFactory().Resource().V1alpha2().ResourceClasses().Lister(),
 		podSchedulingContextLister: fh.SharedInformerFactory().Resource().V1alpha2().PodSchedulingContexts().Lister(),
 		claimParametersLister:      fh.SharedInformerFactory().Resource().V1alpha2().ResourceClaimParameters().Lister(),
+		claimParametersIndexer:     fh.SharedInformerFactory().Resource().V1alpha2().ResourceClaimParameters().Informer().GetIndexer(),
 		classParametersLister:      fh.SharedInformerFactory().Resource().V1alpha2().ResourceClassParameters().Lister(),
+		classParametersIndexer:     fh.SharedInformerFactory().Resource().V1alpha2().ResourceClassParameters().Informer().GetIndexer(),
 		resourceSliceLister:        fh.SharedInformerFactory().Resource().V1alpha2().ResourceSlices().Lister(),
 		claimNameLookup:            resourceclaim.NewNameLookup(fh.ClientSet()),
 		claimAssumeCache:           assumecache.NewAssumeCache(logger, fh.SharedInformerFactory().Resource().V1alpha2().ResourceClaims().Informer(), "claim", "", nil),
 	}
 
+	if err := pl.claimParametersIndexer.AddIndexers(cache.Indexers{generatedFromIndex: claimParametersGeneratedFromIndexFunc}); err != nil {
+		return nil, fmt.Errorf("add claim parameters cache indexer: %w", err)
+	}
+	if err := pl.classParametersIndexer.AddIndexers(cache.Indexers{generatedFromIndex: classParametersGeneratedFromIndexFunc}); err != nil {
+		return nil, fmt.Errorf("add class parameters cache indexer: %w", err)
+	}
+
 	return pl, nil
 }
 
+// claimParametersGeneratedFromIndexFunc is an index function that returns other resource keys
+// (= apiGroup/kind/name) for ResourceClaimParametersReference in a given claim parameters.
+func claimParametersGeneratedFromIndexFunc(obj interface{}) ([]string, error) {
+	parameters, ok := obj.(*resourcev1alpha2.ResourceClaimParameters)
+	if !ok {
+		return []string{}, nil
+	}
+	if parameters.GeneratedFrom == nil {
+		return []string{}, nil
+	}
+	return []string{fmt.Sprintf("%s/%s/%s", parameters.GeneratedFrom.APIGroup, parameters.GeneratedFrom.Kind, parameters.GeneratedFrom.Name)}, nil
+}
+
+// classParametersGeneratedFromIndexFunc is an index function that returns other resource keys
+// (= apiGroup/kind/namespace/name) for ResourceClassParametersReference in a given class parameters.
+func classParametersGeneratedFromIndexFunc(obj interface{}) ([]string, error) {
+	parameters, ok := obj.(*resourcev1alpha2.ResourceClassParameters)
+	if !ok {
+		return []string{}, nil
+	}
+	if parameters.GeneratedFrom == nil {
+		return []string{}, nil
+	}
+	return []string{fmt.Sprintf("%s/%s/%s/%s", parameters.GeneratedFrom.APIGroup, parameters.GeneratedFrom.Kind, parameters.GeneratedFrom.Namespace, parameters.GeneratedFrom.Name)}, nil
```

Comment:
Can you do a quick one-of benchmark whether `parameters.GeneratedFrom.APIGroup + "/" + parameters.GeneratedFrom.Kind + "/" + parameters.GeneratedFrom.Namespace + "/" + parameters.GeneratedFrom.Name` is faster?

Comparing both CPU cycles and memory allocations is interesting.

I doubt that it would show up in real-world benchmarks, but it might be a worthwhile micro-optimization.

Reply:
> Can you do a quick one-of benchmark whether parameters.GeneratedFrom.APIGroup + "/" + parameters.GeneratedFrom.Kind + "/" + parameters.GeneratedFrom.Namespace + "/" + parameters.GeneratedFrom.Name is faster?

you are right. it's the most the best choice. I will update my PR.

```
(base) (⎈|kind-kind:test)➜  ~ go run main.go
'x' + 'y' Concatenate Strings: 41214404	        29.05 ns/op
strings.Join Strings: 28133846	        41.34 ns/op
strings.Builder Strings: 18780309	        63.51 ns/op
fmt.Sprintf Strings:  7718456	       156.1 ns/op
```